fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -313,12 +321,26 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
     }
 
     /**
-     *  Given a FedoraResource, return a list of agents.
+     * Given a FedoraResource, return a list of agents.
      */
     private static Stream<String> getAgentMembers(final IdentifierConverter<Resource, FedoraResource> translator,
-            final FedoraResource resource) {
-        return resource.getTriples(translator, PROPERTIES).filter(memberTestFromTypes.apply(resource.getTypes()))
-            .map(Triple::getObject).flatMap(WebACRolesProvider::nodeToStringStream);
+                                                  final FedoraResource resource, final String hashPortion) {
+
+        //resolve list of triples, accounting for hash-uris.
+        final List<Triple> triples = resource.getTriples(translator, PROPERTIES).filter(
+            triple -> hashPortion == null || triple.getSubject().getURI().endsWith(hashPortion)).collect(toList());
+        //determine if there is a rdf:type vcard:Group
+        final boolean hasVcardGroup = triples.stream().filter(
```

Comment:
I would suggest using `anyMatch` here instead of `count`. Also, just a suggestion, but it may be more readable to use triple.matches:
```
final boolean hasVcardGroup = triples.stream().anyMatch(
                triple -> triple.matches(triple.getSubject(), RDF_TYPE_NODE, VCARD_GROUP_NODE));
```